LibCarla/source/carla/trafficmanager/LocalizationUtils.cpp

```diff
@@ -232,6 +233,43 @@ namespace LocalizationConstants {
     }
 
   }
+  SimpleWaypointPtr TrackTraffic::GetTargetWaypoint(Buffer& waypoint_buffer,const float& target_point_distance,uint64_t& index)
+  {
+    SimpleWaypointPtr target_waypoint = waypoint_buffer.front();
+    uint64_t  startPosn  = static_cast<uint64_t>(std::fabs(target_point_distance/SAMPLING_RESOLUTION));
+    bool mScanForward = false;
+    double target_point_dist_power = std::pow(target_point_distance, 2);
+
+    /// Condition to determine forward or backward scanning of  WayPoint Buffer.
+    if((startPosn < waypoint_buffer.size()) &&
+        (waypoint_buffer.front()->DistanceSquared(target_waypoint)
+        <= target_point_dist_power)) {
+      mScanForward = true;
+    }
+
+    if(mScanForward) {
+      for (uint64_t j = startPosn ;
+          (j < waypoint_buffer.size()) &&
+          (waypoint_buffer.front()->DistanceSquared(target_waypoint)
+          < target_point_dist_power);
+          ++j) {
+          target_waypoint = waypoint_buffer.at(j);
+          index = j;
+          }
+    }
+    else
+    {
+      for (uint64_t j = startPosn ;
+        (j >= 0) &&
+        (waypoint_buffer.front()->DistanceSquared(target_waypoint)
+        < target_point_dist_power);
+        --j) {
+        target_waypoint = waypoint_buffer.at(j);
+        index = j;
+        }
+    }
+  return target_waypoint;
```

Comment:
If the intention of using the index variable was to somehow communicate back to the caller with this identified target index value, then consider creating a local variable instead of an input by reference in the parameter list and return the local variable along with the target_waypoint in a pair.

Reply:
Agreed and implemented

Reply:
LGTM